src/purchases.ts

```diff
@@ -624,6 +624,29 @@ export default class Purchases {
         RNPurchases.syncPurchases();
     }
 
+    /**
+     * This method will send a purchase to the RevenueCat backend. This function should only be called if you are
+     * in Amazon observer mode or performing a client side migration of your current users to RevenueCat.
+     *
+     * The receipt IDs are cached if successfully posted so they are not posted more than once.
+     *
+     * @param {string} productID Product ID associated to the purchase.
+     * @param {string} receiptID ReceiptId that represents the Amazon purchase.
+     * @param {string} amazonUserID Amazon's userID. This parameter will be ignored when syncing a Google purchase.
+     * @param {(string|null|undefined)} isoCurrencyCode Product's currency code in ISO 4217 format.
+     * @param {(number|null|undefined)} price Product's price.
+     * @returns {Promise<void>} The promise will be rejected if configure has not been called yet or if there's an error
+     * syncing purchases.
+     */
+    public static async syncObserverModeAmazonPurchase(productID: string, receiptID: string,
+                                                       amazonUserID: string, isoCurrencyCode?: string | null,
+                                                       price?: number | null): Promise<void> {
+      if (Platform.OS === 'android') {
```

Comment:
I wonder if we should log something if not on Android... Also, right now we have a mix of systems for these single-platform supported methods. Most of them call the unsupported platform wrapper code which just returns without doing anything. Doing something like this makes sense, but we probably would want to clean it up so it's consistent.

Reply:
I agree, I also always hated how differently platforms have different ways this is done. It should be consistent. I noticed how `beginRefundRequestForActiveEntitlement` throws an exception, I will do the same here, since that looks like what we should do, since it's more informative than just not doing anything